collect_app/src/main/java/org/odk/collect/android/activities/AppListActivity.java

```diff
@@ -17,16 +17,180 @@
+            mDrawerToggle.syncState();
+        }
+    }
+
+    @Override
+    public void onConfigurationChanged(Configuration newConfig) {
+        super.onConfigurationChanged(newConfig);
+        if (mDrawerToggle != null) {
+            mDrawerToggle.onConfigurationChanged(newConfig);
+        }
+    }
+
+    private void setupDrawerItems() {
+        ArrayAdapter<String> adapter = new ArrayAdapter<String>(this, android.R.layout.simple_list_item_1, mSortingOptions) {
+            @Override
+            public View getView(int position, View convertView, ViewGroup parent) {
+                TextView textView = (TextView) super.getView(position, convertView, parent);
+                textView.setPadding(50, 0, 0, 0);
+                return textView;
+            }
+        };
+
+        mDrawerList.setAdapter(adapter);
+        mDrawerList.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                switch(position) {
+                    case 0:
+                        sortByNameAsc();
+                        break;
+                    case 1:
+                        sortByNameDesc();
+                        break;
+                    case 2:
+                        sortByDateDesc();
+                        break;
+                    case 3:
+                        sortByDateAsc();
+                        break;
+                    case 4:
+                        sortByStatusAsc();
+                        break;
+                    case 5:
+                        sortByStatusDesc();
+                        break;
+                }
+                mDrawerLayout.closeDrawer(Gravity.RIGHT);
+            }
+        });
+    }
+
+    private void setupDrawer() {
+        mDrawerList = (ListView) findViewById(R.id.sortingMenu);
+        mDrawerLayout = (DrawerLayout) findViewById(R.id.drawer_layout);
+        mDrawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+        mDrawerToggle = new ActionBarDrawerToggle(this, mDrawerLayout, R.string.sorting_menu_open, R.string.sorting_menu_close) {
+            public void onDrawerOpened(View drawerView) {
+                super.onDrawerOpened(drawerView);
+                invalidateOptionsMenu();
+                mDrawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_UNLOCKED);
+            }
+
+            public void onDrawerClosed(View view) {
+                super.onDrawerClosed(view);
+                invalidateOptionsMenu();
+                mDrawerLayout.setDrawerLockMode(DrawerLayout.LOCK_MODE_LOCKED_CLOSED);
+            }
+        };
+
+        mDrawerToggle.setDrawerIndicatorEnabled(true);
+        mDrawerLayout.setDrawerListener(mDrawerToggle);
+    }
+
+    protected void retrieveCheckedItems(List<Integer> checkedInstances, Cursor cursor) {
+        getListView().clearChoices();
+        int id = 0;
```

Comment:
I had a hard time reading through this method because there are so many different IDs! This is really position in the listview, right? I'd call it position or even listViewPosition.